machine/v1beta1/types_controlplanemachineset.go

```diff
@@ -0,0 +1,244 @@
+package v1beta1
```

Comment:
if we won't change this and it will be on by default, it shoudl be v1.  I'm thinking about writing a script to ban new APIs that try to be beta :)

Reply:
> if we won't change this

Now that is a good question, we are currently trying to make this as compatible with Cluster API as possible to support future work there, but, we will definitely want to be adding fields to this in the future and possibly deprecating some of it depending on how the Cluster API project stabilises their (v1beta1) API.

I've written [a bunch about this](https://github.com/openshift/enhancements/pull/1008/files#diff-167c14da636a95920dc7f720fd0d0be7490305b02914e4113ee9aebf2f51a3bdR657-R698) in the enhancement which I thinks we may actually want to evolve this in the future. 

For example, depending on how things go, we may want to move the current `template` into a discriminated union (which I believe could be achieved on upgrade using a conversion webhook) to allow a new `machineTemplate` or some other field to be introduced to support CAPI as well.

I've not really seen how other OpenShift APIs have actually evolved, so I'd be keen to learn more about this and what we are signing ourselves up for with these changes that we are predicting